### PR TITLE
Jw 261 ksync context

### PR DESCRIPTION
**Pull request by mmasztalerczuk**
* Implemented the communication context for ksync channel

### CODE REVIEW

**Comment by mmasztalerczuk** [comment]
Retest this please

File: include/vr_ksync_defs.h [hunk 1]
```diff
@@ -37,6 +37,7 @@ struct ksync_response {
 // TODO: JW-261 Implement multiple queues to handle multiple processes
 //       opening a pipe
```
Comment (m-kostrzewa) [inline include/vr_ksync_defs.h]: We can remove this TODO ;)

File: windows/vr_ksync.c [hunk 2]
```diff
@@ -308,7 +336,7 @@ KsyncDispatchDeviceControl(PDEVICE_OBJECT DriverObject, PIRP Irp)
                 return STATUS_INSUFFICIENT_RESOURCES;
             }
 
-            returnedValue.pBuffer = user_virtual_address;
+            returnedValue.pBuffer = ctx->user_virtual_address;
```
Comment (m-kostrzewa) [inline windows/vr_ksync.c]: Just for reference: http://stackoverflow.com/questions/29938573/accessing-kernel-memory-from-user-mode-windows

File: windows/vr_ksync.c [hunk 3]
```diff
@@ -77,16 +76,40 @@ KsyncDispatchCreate(PDEVICE_OBJECT DriverObject, PIRP Irp)
...
+    {
+        IoCompleteRequest(Irp, IO_NO_INCREMENT);
+        return STATUS_INSUFFICIENT_RESOURCES;
+    }
+
+    PIO_STACK_LOCATION pStack = IoGetCurrentIrpStackLocation(Irp);
+    ctx->user_virtual_address = NULL;
+    ctx->responses = NULL;
+
+    pStack->FileObject->FsContext = ctx;
```
Comment (m-kostrzewa) [inline windows/vr_ksync.c]: Just for reference: http://driverentry.com.br/en/blog/?p=313

**Pull request merged**